js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -315,6 +316,47 @@ export function useLatency(): number | null {
   return latency
 }
 
+type DataChannelMsg = { sender: string; value: any }
+type WebRtcOnMessage = (msg: DataChannelMsg) => void
+
+function useWebRtcBroadcastChannel(throttle = 0) {
+  const db = useDatabase()
+  const id = useUniqueClientId()
+  const WebRtcBroadcastChannelRef = React.useRef<SyncedWebRTCConnections>()
+  if (!WebRtcBroadcastChannelRef.current) {
+    WebRtcBroadcastChannelRef.current = new SyncedWebRTCConnections(db, id, throttle)
+  }
+  return {
+    send: (msg: string) => WebRtcBroadcastChannelRef.current!.send(msg),
+    setOnMessage: (onMessage: WebRtcOnMessage) =>
+      WebRtcBroadcastChannelRef.current!.setOnMessage(onMessage),
+    peers: WebRtcBroadcastChannelRef.current!.peers()
+  }
+}
+
+export function useWebRtcPresence(vals: any, throttle = 0) {
+  const { send, setOnMessage, peers } = useWebRtcBroadcastChannel(throttle)
+  const rtcMap = useRef<Map<string, DataChannelMsg>>()
+  if (!rtcMap.current) rtcMap.current = new Map()
+  const [messageCount, setMessageCount] = useState(0)
+  React.useEffect(() => {
+    send(JSON.stringify(vals))
+  }, [vals])
+  React.useEffect(() => {
+    for (const peer of rtcMap.current!.keys()) {
+      if (![...peers].includes(peer)) rtcMap.current!.delete(peer)
+    }
+  }, [peers])
+  React.useEffect(() => {
+    let count = 0
+    setOnMessage((msg) => {
+      rtcMap.current!.set(msg.sender, msg)
+      setMessageCount(count++)
+    })
+  }, [messageCount])
```

Comment:
I don't think I'm following the logic for why we need to set a new `onMessage` for the WebRTCBroadcastChannel for every message that comes in 🤔